src/server-code/admin-server.js

```diff
@@ -176,22 +172,26 @@ export default function runAdminServer(serverFalcorModel) {
           Key,
         },
       };
-      awsSdkClient.headObject({ Bucket, Key }, (err) => {
-        if (err && err.code === 'NotFound') {
-          const s3Uploader = s3Client.uploadFile(s3Params);
-          s3Uploader.on('error', s3Err => {
-            console.error(s3Err); // eslint-disable-line no-console
-            deleteTmpFile();
-            return res.status(500).send('Error uploading');
-          });
-          s3Uploader.on('end', () => {
-            const imageUrl = s3.getPublicUrl(Bucket, Key);
-            deleteTmpFile();
-            return res.status(200).send(`success ${imageUrl}`);
-          });
-        }
-        deleteTmpFile();
-        return res.status(409).send(`object already exists, ${Key}`);
+
+      /* Compress image, then upload to S3 */
+      compressJPEG(filePath).then(() => {
+        awsSdkClient.headObject({ Bucket, Key }, (err) => {
+          if (err && err.code === 'NotFound') {
+            const s3Uploader = s3Client.uploadFile(s3Params);
+            s3Uploader.on('error', s3Err => {
+              console.error(s3Err); // eslint-disable-line no-console
+              deleteFile(filePath).then(() => {});
```

Comment:
what's with all these `.then`s. a promise starts running as soon as it's created, not when you call `.then` on it.